### PR TITLE
CRUD endpoint for DAG (#9101)

### LINKED ISSUE #9109: API Endpoint - CRUD - DAG
Hello 

We need to create several endpoints that perform basic CRUD operations on **DAG**. We need the following endpoints:

- PATCH /dags/{dag_id}

It depends on https://github.com/apache/airflow/issues/8128

Detailed information is available in the issue:
https://github.com/apache/airflow/issues/8118

LOVE,

### PR DESCRIPTION
**Pull request by OmairK**
Closes #9109 
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: tests/api_connexion/endpoints/test_dag_endpoint.py [hunk 1]
```diff
@@ -255,12 +256,123 @@ def test_should_response_200_default_limit(self):
...
+                        "years": 0,
+                    },
+                    "tags": [],
+                },
+            ),
+        ]
+    )
+    @provide_session
+    def test_should_response_200(self, name, patch_body, expected_reponse, session):
+        del name
```
Comment (jhtimmins) [inline tests/api_connexion/endpoints/test_dag_endpoint.py]: Looks like the pattern for passing in an unused name variable, according to the docs, is to rename `name` as `_`. Then you don't have to delete the value.

Comment (OmairK) [reply tests/api_connexion/endpoints/test_dag_endpoint.py]: Neat :smile:! Fixed it.

**Comment by mik-laj** [comment]
I updated this branch and made some minor changes - the schedule_interval field is read-only.

**Comment by OmairK** [comment]
> I updated this branch and made some minor changes - the schedule_interval field is read-only.

@mik-laj thank you, are there any changes left that I should do or is this PR good to go?

**Comment by mik-laj** [comment]
@OmairK nothing left, but not merged yet, because I waited for confirmation that this change looks good for you.

**Review by mik-laj — APPROVED** [review]

**Pull request merged**